microsoft/testsuites/display/drm.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from typing import Any, cast
+
+from assertpy import assert_that
+
+from lisa import (
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+
+from lisa.sut_orchestrator import AZURE, READY
+from lisa.base_tools import Cat
+from lisa.tools import Lsmod, Uname
+from lisa.util import SkippedException
+
+@TestSuiteMetadata(
+    area="drm",
+    category="functional",
+    description="""
+    This test suite uses to verify drm driver sanity.
+    """,
+    requirement=simple_requirement(supported_platform_type=[AZURE, READY]),
+)
+class Drm(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        This case is to check whether the hyperv_drm driver registered successfully.
+        Once driver is registered successfully it should appear in `lsmod` output.
+
+        Steps,
+        1. lsmod
+        2. Check if hyperv_drm exist in the list.
+        """,
+        priority=2,
+        use_new_environment=True,
+    )
+    def verify_drm_driver(self, node: Node, log: Logger) -> None:
+        lsmod = node.tools[Lsmod]
+        assert_that(lsmod.module_exists("hyperv_drm")).described_as('hyperv_drm module is absent').is_equal_to(True)
+
+    @TestCaseMetadata(
+        description="""
+        This case is to check whether the dri node is populated correctly.
+        If hyperv_drm driver is bind correctly it should populate dri node.
+        This dri node can be find at following sysfs entry : /sys/kernel/debug/dri/0.
+        The name of this dri node (/sys/kernel/debug/dri/0/name) should contain `hyperv_drm`.
+
+        Step,
+        1. Cat /sys/kernel/debug/dri/0/name.
+        2. Verify it contains hyperv_drm string in it.
+        """,
+        priority=2,
+    )
+    def verify_dri_node(self, node: Node, log: Logger) -> None:
+        cat = node.tools[Cat]
+
+        dri_path = "/sys/kernel/debug/dri/0/name";
+        dri_name = cat.read(dri_path, sudo=True, force_run=True)
+        assert_that(dri_name).described_as('dri node not populated for hyperv_drm').matches("hyperv_drm")
+
+    def before_case(self, node: Node, log: Logger, **kwargs: Any) -> None:
+        kernel_version = node.tools[Uname].get_linux_information().kernel_version
+        if kernel_version <= "5.14.0":
+            raise SkippedException("DRM hyperv driver is supported from 5.14.x kernel.")
```

Comment:
If it's supported *from* 5.14, should the condition be `if kernel_version < "5.14.0":`?

Reply:
I observed that even kernel_version < "5.14.0" is not fully correct, as it skips "5.14.0-rc*" versions which should have supported. I decided to skip 5.14.0 all together for consistency , but revisiting it I think  kernel_version <= "5.13.0" will be the best condition.